src/mozmlops/examples/image_classifier/README.md

```diff
@@ -0,0 +1,25 @@
+# Example: Image Classifier
```

Comment:
Let's think how to make the examples discoverable: we should consider at least mentioning in the top-level readme that we do have examples and how to reach them.

Reply:
....you know what? 

The more I think about it (@Dexterp37 please don't hate me) the more I think it _would_ be prudent to set up a separate examples repository. 

WAIT! Let me explain! 

1. Every time an internal client has approached me about operationalizing their models, the very first thing they ask for is access to other clients' repositories so they can see examples. They care about this _much_ more than they care about the code of our library (which makes sense). It would be nice for them to be able to clone that without having to clone the library we're asking them to import. 

2. I think, particularly over time, the cadence of updates on `mozmlops` will differ from the cadence of updates on examples. In particular, I'd love to be adding examples on a _continuous basis_, particularly as we get rolling with convincing teams to use this toolset for lots of different types of models. I'd love for us to be adding examples often enough that we should _decouple_ the library release cadence from the example addition cadence. 

Maybe in this case we actually just _remove_ the templates from the library. I suspect people would rather copy and paste into the working example that most closely matches their use case, which ideally in my vision they have _options_ for, than generate a new flow from a template that matches their use case _less_. The template _could_ go into the examples repo as a demonstration of a classical machine learning model run on tabular data, though.

Reply:
I think you raised two excellent points. For instance, the repo-based approach is what [torch](https://github.com/pytorch/examples/) uses (and some other bigger repos).

I'm not against this, as long as we time-box it and configure it properly

Reply:
@chelseatroy I have few questions inlined below for my general understanding:

>     1. Every time an internal client has approached me about operationalizing their models, the very first thing they ask for is access to other clients' repositories so they can see examples (which I can't even give them, because I don't have access to the Content ML or Fakespot repos, and revenue and search are both very limited-access). It would be nice for clients to be able to clone an example repo full of examples for them to choose from, without having to clone the library we're asking them to import.
> 

What could go wrong if clients clone the mozmlops library for examples? If we keep examples on top level folder as @Dexterp37 mentioned, they will not be a part of the library anyway. Can the cloned mozmlops conflict with the one imported in Flows?

>     2. I think, particularly over time, the cadence of updates on `mozmlops` will differ from the cadence of updates on examples. In particular, I'd love to be adding examples on a _continuous basis_, particularly as we get rolling with convincing teams to use this toolset for lots of different types of models. I'd love for us to be adding examples often enough that we should _decouple_ the library release cadence from the example addition cadence.
> 

I am 100% with you on adding examples oftener and also agree that the cadence of updates on `mozmlops` might differ from the cadence of updates on examples. But how does keeping examples in this repo affect the library release cadence? I mean we can release the new version of the library independent of what happens with examples in this repo. Right?


> I'm not against this, as long as we time-box it and configure it properly

@Dexterp37 What timeline do you have in mind and what do you mean by `configure`? 


### Based on the overall discussion above, I see 2 different topics

#### Where to add examples
The options are:
1. In the top level folder of this repo so that it doesn't form the part of the library that clients import, OR
2. In a new repo (say `examples` repo)

I am more inclined to go with 1st option here (my answer can change based on @chelseatroy 's response to my queries above) mainly because the pre-requisites to run the examples are [documented](https://github.com/mozilla/mozmlops/blob/main/src/mozmlops/templates/README.md#training-and-deploying-models-in-the-cloud) in this repo. Having a separated example repo means the clients will have to be redirected to this repo (yes, we can link to this documentation from the new repo) but it seems a bit counter-intuitive to me.


#### What do we do with templates
1. Remove them altogether
2. Move them with examples wherever they reside

This one requires more discussions and some work, because:
1. We plan to add new templates for Ray Serve in H2. So, removing the templates will require more discussion.
2. Irrespective of what option we choose, I believe we will have to update our [README](https://github.com/mozilla/mozmlops/blob/main/src/mozmlops/templates/README.md#training-and-deploying-models-in-the-cloud) as right now it is tightly coupled with the [template](https://github.com/mozilla/mozmlops/blob/main/src/mozmlops/templates/template_flow.py) in this repository.

**I would like to tackle the question around templates in a separate issue as it is out of scope of the current task.**

Reply:
> > I'm not against this, as long as we time-box it and configure it properly
> 
> @Dexterp37 What timeline do you have in mind and what do you mean by `configure`?

I mean at the very least [this](https://mozilla-hub.atlassian.net/wiki/spaces/DATA/pages/6848799/GitHub+practices). On the top of that, likely CI tasks and ongoing maintenance.

This effort needs to be timeboxed to this week (ending today).
 
> **I would like to tackle the question around templates in a separate issue as it is out of scope of the current task.**

Yes, I believe the 'templates' discussion is orthogonal and should not be tackled here, but further discussed in the next team meeting.

Reply:
As this needs to be done by EOD today, I am moving out the examples from `src/mozmlops` and landing it in top level directory in this repository.

Any work related to templates and a separate examples repo (if we decide so), will be tackled via a separate ticket.